lib/Controller/OpenAiAPIController.php

```diff
@@ -157,4 +160,53 @@ public function getImageGenerationPage(string $hash): TemplateResponse {
 		$this->initialStateService->provideInitialState('generation', $generationData);
 		return new TemplateResponse(Application::APP_ID, 'imageGenerationPage');
 	}
+
+	/**
+	 * @param bool|null $clearTextPrompts
+	 * @param bool|null $clearImagePrompts
+	 * @return DataResponse
+	 */
+	#[NoAdminRequired]
+	public function clearPromptHistory(?bool $clearTextPrompts = null, ?bool $clearImagePrompts = null): DataResponse {
+		$this->logger->debug('Clearing prompt history: ' . 'clearTextPrompts ' . strval($clearTextPrompts) . ' clearImagePrompts ' . strval($clearImagePrompts));
+		if ($clearTextPrompts === True) {			
+			try {
+				$this->openAiAPIService->clearPromptHistory($this->userId, Application::PROMPT_TYPE_TEXT);
+			} catch (Exception $e) {				
+				return new DataResponse(['error' => $e->getMessage()],Http::STATUS_BAD_REQUEST);
```

Comment:
I'd say a Database exception is rather a 500 than a 400, no?

Reply:
I changed the error handling quite a bit; do you think it would be worthwhile to add the proper code to each exception? I get that WAFs can filter traffic based on the response code but 400 should be quite innocuous in that sense...? Other than that the consumers of the routes will mostly be this app itself, especially with the providers being implemented.